### PR TITLE
Task js dom

### PR DESCRIPTION
**Pull request by alstep07**
### JS DOM

[Demo](https://alstep07.github.io/task-js-dom/) |
[Code base](https://github.com/alstep07/alstep07.github.io/tree/master/task-js-dom)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/alstep07/task-js-dom/app.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+});
+
+function render(genre){
+
+    if (main.hasChildNodes()){
+        main.firstChild.remove();
+    }
+
+    let filmCards = document.createElement('div');
+    filmCards.className="main__genre";
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:284]: Use [`classList`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) it is more predictable then `className`.

File: submissions/alstep07/task-js-dom/app.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+        let filmData = document.createElement('div');
+        let filmDescr = document.createElement('p');
+        let card = document.createElement('div');
+        let cover = document.createElement('img');
+        let title = document.createElement('h2');
+        let year = document.createElement('p');
+        let rating = document.createElement('p')
+        let descr = document.createElement('p');
+        let director = document.createElement('p');
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:296]: `const`

File: submissions/alstep07/task-js-dom/app.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+        cover.setAttribute('src',`./images/${genre+index}.jpg`);
+        cover.setAttribute('alt', 'cover');
+
+        filmDescr.className = 'film__descr';
+        cover.className ='film__cover film__cover-active';
+        title.className ='film__title';
+        year.className ='film__year';
+        rating.className ='film__rating';
+        director.className ='film__director';  
+        filmData.className='film__data film__data-active';
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:307]: [`classList`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)

File: submissions/alstep07/task-js-dom/app.js [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+        
+        title.textContent=`${film['title']}`;
+        year.textContent=`year: ${film['year']}`;
+        rating.textContent=`imdb: ${film['rating']}`;
+        filmDescr.textContent=`${film['description']}`;
+        director.textContent = `${film['director']}`;
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:313]: You can use dot notation. `film.title`. You don't need bracket notation here.

File: submissions/alstep07/task-js-dom/app.js [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+        
+        filmData.appendChild(title);
+        filmData.appendChild(year);
+        filmData.appendChild(rating);
+        filmData.appendChild(director);
+        card.appendChild(cover);
+        card.appendChild(filmData);
+        card.appendChild(filmDescr);
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:321]: With [append](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append) you can do it in one operation.
`filmData.append(el1, el2, ...)`

File: submissions/alstep07/task-js-dom/app.js [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+        let filmDescr = document.createElement('p');
+        let card = document.createElement('div');
+        let cover = document.createElement('img');
+        let title = document.createElement('h2');
+        let year = document.createElement('p');
+        let rating = document.createElement('p')
+        let descr = document.createElement('p');
+        let director = document.createElement('p');
+        
+        cover.setAttribute('src',`./images/${genre+index}.jpg`);
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/app.js:298]: It is not safe to use string concatenation here. You can implement this logic in such way - `${genre}${index}`

File: submissions/alstep07/task-js-dom/index.html [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      <h2 class="navbar__header">By Genre</h2>
+      <ul class="navbar__list">
+        <li ><a href="#" class="navbar__item">Action</a></li>
+        <li ><a href="#" class="navbar__item">Drama</a></li>
+        <li ><a href="#" class="navbar__item">Comedy</a></li>
+        <li ><a href="#" class="navbar__item">Western</a></li>
+        <li ><a href="#" class="navbar__item">Romance</a></li>
+        <li ><a href="#" class="navbar__item">Horror</a></li>
+      </ul>
+    </nav>
```
Comment (babayK0) [inline submissions/alstep07/task-js-dom/index.html:26]: Are you try to use [`bem`](https://ru.bem.info/methodology/quick-start/) methodology ?

I think `navbar__item`'s  are `h2` and `ul` tags only.

One more point. Maybe try to render this list with js ?)

**Review by babayK0 — CHANGES_REQUESTED** [review]
Good job ! 

Lets try to move your data (movies) in another file. You can use [JavaScript modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for this

Also watch my comments below)

**Comment by babayK0** [comment]
@alstep07 why you close pull request ?

**Comment by alstep07** [comment]
> @alstep07 why you close pull request ?



> @alstep07 why you close pull request ?

I'm sorry for this.
I'm totally confused with merging two last tasks. I try to delete all changes and restart

**Comment by alstep07** [comment]
can

**Pull request closed without merging**